src/wallet/wallet.h

```diff
@@ -763,20 +766,42 @@ enum class PrivacyPolicy {
     NoPrivacy,
 };
 
+/** Returns the meet of two privacy policies. I.e., the strongest policy that is
+ *  compatible with both of the provided policies.
+ *
+ *  See https://github.com/zcash/zcash/issues/6240 for the graph that this
+ *  models.
+ */
+PrivacyPolicy PrivacyPolicyMeet(PrivacyPolicy a, PrivacyPolicy b);
+
 class TransactionStrategy {
-    PrivacyPolicy privacy;
+    PrivacyPolicy requestedLevel;
 
 public:
-    TransactionStrategy() : privacy(PrivacyPolicy::FullPrivacy) {}
-    TransactionStrategy(const TransactionStrategy& strategy) : privacy(strategy.privacy) {}
-    TransactionStrategy(PrivacyPolicy privacyPolicy) : privacy(privacyPolicy) {}
+    TransactionStrategy() : requestedLevel(PrivacyPolicy::FullPrivacy) {}
+    TransactionStrategy(const TransactionStrategy& strategy) : requestedLevel(strategy.requestedLevel) {}
+    TransactionStrategy(PrivacyPolicy privacyPolicy) : requestedLevel(privacyPolicy) {}
 
     static std::optional<TransactionStrategy> FromString(std::string privacyPolicy);
+    static std::string ToString(PrivacyPolicy policy);
+
+    std::string PolicyName() const {
+        return ToString(requestedLevel);
+    }
 
-    bool AllowRevealedAmounts();
-    bool AllowRevealedRecipients();
-    bool AllowRevealedSenders();
-    bool AllowLinkingAccountAddresses();
+    bool AllowRevealedAmounts() const;
+    bool AllowRevealedRecipients() const;
+    bool AllowRevealedSenders() const;
+    bool AllowFullyTransparent() const;
+    bool AllowLinkingAccountAddresses() const;
+
+    // A strategy is compatible with a given required level if
+    // it is as strong as, or weaker than, the required level.
```

Comment:
```suggestion
    // A strategy is compatible with a given required level if it is
    // either exactly as strong as, or weaker than, the required level.
```

Reply:
It's not 100% clear what "weaker than" means here. It's obvious what it would mean for a strict order, but for a partial order there are two possible ways it could be interpreted, I think (the issue being what we return for the case where the requested and required levels are incomparable).

Reply:
“weaker than” just means `≤`, so incomparable ones return `false`. The “as strong as” part is redundant, but figured it clarified the colloquial interpretation of “weaker than” as “strictly weaker than”. However, I’ll reword things to be closer to the poset terminology throughout.